near/omni-bridge/src/lib.rs

```diff
@@ -196,53 +201,35 @@ impl FungibleTokenReceiver for Contract {
         amount: U128,
         msg: String,
     ) -> PromiseOrValue<U128> {
-        let parsed_msg: InitTransferMsg = serde_json::from_str(&msg).sdk_expect("ERR_PARSE_MSG");
         let token_id = env::predecessor_account_id();
-        require!(
-            parsed_msg.recipient.get_chain() != ChainKind::Near,
-            "ERR_INVALID_RECIPIENT_CHAIN"
-        );
-
-        self.current_origin_nonce += 1;
-        let destination_nonce = self.get_next_destination_nonce(parsed_msg.recipient.get_chain());
-
-        let transfer_message = TransferMessage {
-            origin_nonce: self.current_origin_nonce,
-            token: OmniAddress::Near(token_id.clone()),
-            amount,
-            recipient: parsed_msg.recipient,
-            fee: Fee {
-                fee: parsed_msg.fee,
-                native_fee: parsed_msg.native_token_fee,
-            },
-            sender: OmniAddress::Near(sender_id.clone()),
-            msg: String::new(),
-            destination_nonce,
+        let parsed_msg: BridgeOnTransferMsg =
+            serde_json::from_str(&msg).sdk_expect("ERR_PARSE_MSG");
```

Comment:
Let's try to parse in the old format if the new one fails.

Reply:
e67f0bcf9c67375666e359e34e808545ae7328d8